### PR TITLE
add tests for process models

### PR DESCRIPTION
**Pull request by nwalsh1995**
Cleans up the logic and also adds a small test using mocks.

@osrsbox, I made some TODOs in the code that I think you can tackle when you have time. 

One thing is that I would like to have some better data for the models like we have for the items. Right now we are mocking a lot of the returns so there are chances for something to go wrong. But as it stands, we have sufficient test coverage of this file.

#18

### CODE REVIEW

**Comment by osrsbox** [comment]
Thanks for updating this module. It was interesting to read your code and see you approach. There was a lot of simplification, it reads well. Some of my original variable names not very descriptive. I would be happy to change them, but would need collaborator access to your fork - or we could just merge then I make the changes.

I also rewrote the `main` method to hopefully be a little more elegant! And posted a potential solution in the review about getting the data for a thorough test.

File: test/test_process_models.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from unittest.mock import MagicMock, patch, mock_open
+
+import pytest
+
+
+# @TODO(PH01L): We should get a version of this that uses live data instead of mocks.
+@pytest.mark.parametrize("json_loads_return", [
+    {"id": "id", "name": "name", "type": "items", "inventoryModel": ["model"]},
+    {"id": "id", "name": "name", "type": "items", "models": ["model"]},
+    {"id": "id", "name": "name", "type": "items", "objectModels": ["model"]},
```
[the remark this answers is not in the document]
Comment (nwalsh1995) [reply test/test_process_models.py]: Oops, I forgot to change those! Nice catch.

File: model_db_tools/ProcessModels.py [hunk 2]
```diff
@@ -47,88 +47,62 @@
...
-    objects = os.path.join(directory, "objects", "")
-
-    models_dict = dict()
+def main(path_to_definitions):
+    items = os.path.join(path_to_definitions, "items", "")
+    npcs = os.path.join(path_to_definitions, "npcs", "")
+    objects = os.path.join(path_to_definitions, "objects", "")
+    models_dict = {}
 
+    # TODO(@PH01L): The following 3 for loops can be simplified since they share common code.
```
[the remark this answers is not in the document]
Comment (nwalsh1995) [reply model_db_tools/ProcessModels.py]: Nice I was thinking very similar. The only thing that I will change from your code is the key:
```python
key = model["type"] + "_" + str(model["type_id"]) + "_" + str(model["model_id"])  # old
key = f"{model['type']}_{model['type_id']}_{model['model_id']}"  # new
```

**Comment by nwalsh1995** [comment]
Addressed your comments and made sure the new changes pass our tests.

**Pull request merged**